kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h

```diff
@@ -853,12 +850,10 @@ class ResidualBasedBlockBuilderAndSolver
         BaseType::mEquationSystemSize = BaseType::mDofSet.size();
         int ndofs = static_cast<int>(BaseType::mDofSet.size());
 
-        #pragma omp parallel for firstprivate(ndofs)
-        for (int i = 0; i < static_cast<int>(ndofs); i++) {
-            typename DofsArrayType::iterator dof_iterator = BaseType::mDofSet.begin() + i;
-            dof_iterator->SetEquationId(i);
-
-        }
+        IndexPartition<std::size_t>(ndofs).for_each([&](std::size_t Index){

```

Comment:
same

Reply:
I think we cant use block_for_each() here, Hence keeping IndexPartition()